test/js/reducers/questionnaire.spec.js

```diff
@@ -816,6 +813,32 @@ describe('questionnaire reducer', () => {
       Exercises: ['Yes', 'No']
     })
   })
+
+  describe('csv for translation', () => {
```

Comment:
This test covers almost all that is needed to check the CSV Download works. The only piece missing is to add rows for the language selection prompts to the `expected` array. 

Currently this test fails. That is because the `state.data` fed to the `csvForTranslation` function is missing prompts in Spanish, which are important to validate the CSV export procedure (otherwise you'd rows with only English values in the first column).

I had originally added prompts to the dummy questionnaire at the bottom of this file, but that broke all the tests related to language management once I tried to rebase from `master`.

Then I rollbacked the changes to the dummy questionnaire and tried to get to the same state by issuing `addLanguage` actions and corresponding "prompt change" actions. 

When I set out to do this I discovered that `addLanguage` and `setDefaultLanguage` only affect the questionnaire's `languages` and `defaultLanguage` properties while they should also fill the model with properly initialized data for modes, prompts and languages. 

That caused the validation code to break in some places, because it's expecting that when the questionnaire has a given mode enabled, the structures underneath prompts and responses are sound.

I guess all of these wasn't uncovered until now because all the validation code is pre-multilingual support and assumes that everything is under a hardcoded 'en' key.